src/main/java/seedu/storage/Csv.java

```diff
@@ -0,0 +1,207 @@
+                writer.append(String.join(",", fields)).append("\n");
+            }
+
+            // Handle records
+            for (Map<String, String> record : inventory.getRecords()) {
+                for (int i = 0; i < fields.size(); i++) {
+                    String field = fields.get(i);
+                    String value = record.getOrDefault(field, "null");
+                    writer.append(value != null ? value : "null");
+                    if (i < fields.size() - 1) {
+                        writer.append(",");
+                    }
+                }
+                writer.append("\n");
+            }
+            System.out.println("CSV file updated after deletion, including metadata.");
+        } catch (IOException e) {
+            System.err.println("Error updating CSV after deletion: " + e.getMessage());
+        }
+    }
+
+    // Update the CSV file headers and keep the existing data
+    public void updateCsvHeaders(Inventory inventory) {
+        List<String> fields = inventory.getFields();
+        Map<String, String> fieldTypes = inventory.getFieldTypes();
+
+        // metadata include for persistent field type
+        List<String> metadata = new ArrayList<>();
+        for (String field : fields) {
+            String type = fieldTypes.get(field);
+            metadata.add(field + ":" + type);
+        }
+        String metadataLine = "#" + String.join(",", metadata);
+
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(csvFilePath))) {
+            // Write metadata line
+            writer.write(metadataLine);
+            writer.newLine();
+            // Write header line (field names)
+            writer.write(String.join(",", fields));
+            writer.newLine();
+            // Write data records
+            for (Map<String, String> record : inventory.getRecords()) {
+                List<String> values = new ArrayList<>();
+                for (String field : fields) {
+                    values.add(record.getOrDefault(field, ""));
+                }
+                writer.write(String.join(",", values));
+                writer.newLine();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // Append a record to the CSV file considering the field order
+    public void appendRecord(Map<String, String> record, Inventory inventory) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(csvFilePath, true))) {
+            List<String> values = new ArrayList<>();
+            for (String field : inventory.getFields()) {
+                values.add(record.getOrDefault(field, ""));
+            }
+            writer.write(String.join(",", values));
+            writer.newLine();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // Load records from the CSV file into the Inventory
+    public void loadInventoryFromCsv(Inventory inventory) {
```

Comment:
This is a very long method; consider breaking the method up into smaller pieces.